### PR TITLE
checker(dm): add downstream CDC/PiTR check for lightning

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #3510

### What is changed and how it works?

Also I remove a local disk checking in this PR, because precheck is performed at DM-master and the lightning is run at DM-worker.

#### Tests 

 - Manual test (add detailed scripts or steps below)
 ```
                        {
                                "id": 28,
                                "name": "lightning_downstream_cdc_pitr",
                                "desc": "check whether the downstream has tasks incompatible with Physical import mode",
                                "state": "fail",
                                "errors": [
                                        {
                                                "severity": "fail",
                                                "short_error": "found CDC capture(s): clusterID: default captureID(s): [12e1ad9b-b461-48ba-83d8-9dc4d226001b b5def693-5849-4a54-974a-52fa34869273],\nlocal backend is not compatible with them. Please switch to tidb backend then try again."
                                        }
                                ],
                                "instruction": "you can switch to logical import mode which has no requirements on this"
                        }
```

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
also check the effects of `sorting-dir-physical`

File: dm/pkg/checker/lightning.go [hunk 1]
```diff
@@ -205,40 +205,40 @@ func (c *LightningFreeSpaceChecker) Check(ctx context.Context) *Result {
...
-type LightningSortingSpaceChecker struct {
+// LightningCDCPiTRChecker checks whether the cluster has running CDC PiTR tasks.
+type LightningCDCPiTRChecker struct {
 	inner restore.PrecheckItem
 }
 
-// NewLightningSortingSpaceChecker creates a new LightningSortingSpaceChecker.
-func NewLightningSortingSpaceChecker(lightningChecker restore.PrecheckItem) RealChecker {
-	return &LightningSortingSpaceChecker{inner: lightningChecker}
+// NewLightningCDCPiTRChecker creates a new LightningClusterVersionChecker.
```
Comment (D3Hunter) [inline dm/pkg/checker/lightning.go:213]:
```suggestion
// NewLightningCDCPiTRChecker creates a new LightningCDCPiTRChecker.
```

File: dm/pkg/checker/lightning.go [hunk 2]
```diff
@@ -205,40 +205,40 @@ func (c *LightningFreeSpaceChecker) Check(ctx context.Context) *Result {
...
+func NewLightningCDCPiTRChecker(lightningChecker restore.PrecheckItem) RealChecker {
+	c, ok := lightningChecker.(*restore.CDCPITRCheckItem)
+	if ok {
+		c.Instruction = "physical import mode is not compatible with them. Please switch to logical import mode then try again."
```
Comment (D3Hunter) [inline dm/pkg/checker/lightning.go:217]:
```suggestion
		c.Instruction = "there are running CDC or PiTR, physical import mode is not compatible with them. Please switch to logical import mode then try again."
```

File: dm/config/checking_item.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ const (
 	LightningRegionDistributionChecking = "region_distribution"
 	LightningDownstreamVersionChecking  = "downstream_version"
 	LightningFreeSpaceChecking          = "free_space"
-	LightningSortingSpaceChecking       = "enough_sorting_space"
```
Comment (D3Hunter) [inline dm/config/checking_item.go:45]: not needed?

Comment (buchuitoudegou) [reply dm/config/checking_item.go:45]: seem duplicate with `free_space`

File: dm/pkg/checker/lightning.go [hunk 4]
```diff
@@ -62,7 +62,7 @@ func (c *LightningEmptyRegionChecker) Name() string {
 func (c *LightningEmptyRegionChecker) Check(ctx context.Context) *Result {
 	result := &Result{
 		Name:  c.Name(),
-		Desc:  "check whether there are too many empty Regions in the TiKV under Physical import mode",
+		Desc:  "check whether there are too many empty Regions in the TiKV under physical import mode",
```
Comment (buchuitoudegou) [inline dm/pkg/checker/lightning.go:65]:
```suggestion
		Desc:  "check whether there are too many empty regions in the TiKV under physical import mode",
```

**Review by buchuitoudegou — APPROVED** [review]
rest lgtm

File: dm/pkg/checker/lightning.go (hunk 2 above)
Comment (lance6716) [reply dm/pkg/checker/lightning.go:217]: The detailed message (CDC or PiTR) is printed at lightning checker's side, like 
```
"found PiTR log streaming task(s): [br_name],\n"+
		"found CDC capture(s): clusterID: <nil> captureID(s): [f14cb04d-5ba1-410e-a59b-ccd796920e9d],\n"+
		"local backend is not compatible with them. Please switch to tidb backend then try again."
```
no need to repeat again

File: dm/config/checking_item.go (hunk 3 above)
Comment (lance6716) [reply dm/config/checking_item.go:45]: "sorting space" is a property of DM worker, but precheck is done at DM master. I forget this limit

**Comment by lance6716** [comment]
ptal @D3Hunter

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**